torch/_inductor/codegen/triton.py

```diff
@@ -849,7 +850,11 @@ def expm1(x):
 
     @staticmethod
     def sqrt(x):
-        return f"libdevice.sqrt({x})"
+        needs_upcast = x.dtype in (torch.float16, torch.bfloat16)
+        orig_dtype = str(x.dtype).replace("torch", "tl")
```

Comment:
There is a method for this elsewhere in this file, please reuse it. `triton_type`

Reply:
Right now it seems that `triton_type` will map float16 / bfloat16 to float32, which is not the original input dtype.

Reply:
That's what `triton_compute_type` does. not `triton_type`.